### PR TITLE
Setup Storage testing

### LINKED ISSUE #1186: Setup the testing flow for Storage
- setup cypress
- setup the first login flow

### PR DESCRIPTION
**Pull request by Tbaut**
closes #1186

Files and Storage test will only run if some files have been touched that include the directories they are using.
In this PR for instance I do touch the root `package.json`, so both test suit will run. 
image

### CODE REVIEW

File: packages/storage-ui/cypress/support/page-objects/basePage.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Only add things here that could be applicable to all / most pages
+
+export const basePage = {
+  cidsHeaderLabel: () => cy.get("[data-cy=cids-header]", { timeout: 20000 }),
```
Comment (asnaith) [inline packages/storage-ui/cypress/support/page-objects/basePage.ts]: I don't think these elements should be part of the basePage object as they are not on all pages but all good to leave them here for now and I can move later once we have the cidsPage and bucketsPage objects 🙂

**Review by asnaith — APPROVED** [review]
Wow @Tbaut this is really awesome 🤩! You got this in place quickly. The configuration so that the tests only run when necessary is very nice too. Thanks for this 🙏

File: .github/workflows/test-storage.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          REACT_APP_BLOCKNATIVE_ID: ${{ secrets.GH_REACT_APP_BLOCKNATIVE_ID }}
+          REACT_APP_FILES_VERIFIER_NAME: ${{ secrets.GH_REACT_APP_FILES_VERIFIER_NAME }}
+          REACT_APP_FILES_UUID_VERIFIER_NAME: 'chainsafe-uuid-testnet'
+          REACT_APP_TEST: 'true'
+        with:
+          start: yarn start:storage-ui
+          # quote the url to be safe against YML parsing surprises
+          wait-on: 'http://localhost:3000'
+          # wait for max 2 minutes for the storage-ui to respond
+          wait-on-timeout: 120
```
Comment (Tbaut) [inline .github/workflows/test-storage.yml:46]: yolo
```suggestion
          wait-on-timeout: 240
```

File: .github/workflows/test-storage.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          REACT_APP_BLOCKNATIVE_ID: ${{ secrets.GH_REACT_APP_BLOCKNATIVE_ID }}
+          REACT_APP_FILES_VERIFIER_NAME: ${{ secrets.GH_REACT_APP_FILES_VERIFIER_NAME }}
+          REACT_APP_FILES_UUID_VERIFIER_NAME: 'chainsafe-uuid-testnet'
+          REACT_APP_TEST: 'true'
+        with:
+          start: yarn start:storage-ui
+          # quote the url to be safe against YML parsing surprises
+          wait-on: 'http://localhost:3000'
+          # wait for max 2 minutes for the storage-ui to respond
+          wait-on-timeout: 240
```
Comment (Tbaut) [inline .github/workflows/test-storage.yml:46]:
```suggestion
          wait-on-timeout: 120
```

File: packages/storage-ui/cypress/support/page-objects/basePage.ts (hunk 1 above)
Comment (Tbaut) [reply packages/storage-ui/cypress/support/page-objects/basePage.ts]: thanks, I changed that

**Comment by Tbaut** [comment]
hmm I'm starting to think that it's just Github action playing against me.. there's no video uploaded on the `arg` commit https://github.com/ChainSafe/files-ui/runs/2909029464
```
Run actions/upload-artifact@v1
  with:
    name: cypress-videos
    path: packages/storage-ui/cypress/videos
  env:
    CI: 1
    CYPRESS_CACHE_FOLDER: /github/home/.cache/Cypress
    npm_config_cache: /github/home/.npm
    TERM: xterm
Uploading artifact 'cypress-videos' from '/home/runner/work/files-ui/files-ui/packages/storage-ui/cypress/videos' for run #2563
Uploading 0 files
File upload complete.
Uploaded '0' bytes from '/home/runner/work/files-ui/files-ui/packages/storage-ui/cypress/videos' to server
Associated artifact cypress-videos (15250779) with run #2563
```

**Comment by Tbaut** [comment]
I'm helpless at this point.. The video shows that there's a white screen as if the server was never built.

**Review by RyRy79261 — APPROVED** [review]

**Comment by Tbaut** [comment]
it's geeeen ! you saved the day @asnaith thank you so much!

**Pull request merged**